### DISCUSSION TITLE
podman remote build with a local Containerfile.

### DISCUSSION THREAD

**Discussion by kiranaaditya** [containers/podman#18895] [Ideas]
I am running a podman remote setup on my windows laptop connecting to a linux server remotely. When I am running the podman build command, it is taking the Dockerfile present in the home directory of the remote machine by default. I went through the docs for podman build and I came across this:

> Local directory – e.g. --build-context project2=../path/to/project2/src (This option is not available with the remote Podman client. On Podman machine setup (i.e macOS and Winows) path must exists on the machine VM)

Why not create an option to build on Dockerfiles from the localhost itself. podman definitely has the permissions for reading the files in the remote host and running.

It would be a simple task of transporting the file over the ssh tunnel, creating a new Containerfile and running it, assuming podman has the permissions to do that. I wanted to know the constraints on this, if it is not possible.

**Comment by Luap99** [comment]
The normal build context is always read on the client side and send as tar to the server. A podman-remote build /path/to /directroy will send the whole directory to the server. Inside the directory you need to have the Dockerfile or Containerfile.

**Reply by kiranaaditya** [reply]
Thanks so much your help. This newbie is happy. Live long and prosper. :)
